File: split_cats.py
import sys
import os
from PIL import Image

input_path = '/Users/simbosung/Downloads/파일이름.png'
out_dir = 'public/emotions'

if not os.path.exists(out_dir):
    os.makedirs(out_dir)

try:
    img = Image.open(input_path).convert("RGBA")
except Exception as e:
    print(f"Error opening image: {e}")
    sys.exit(1)

width, height = img.size

# Divide height by 3
h1 = height // 3

cats = [
    img.crop((0, 0, width, h1)),
    img.crop((0, h1, width, h1 * 2)),
    img.crop((0, h1 * 2, width, height))
]

names = ['angry.png', 'sad.png', 'happy.png']

def make_transparent(image):
    pixels = image.load()
    w, h = image.size
    
    # BFS to find all background pixels
    visited = set()
    queue = []
    
    # Add edges
    for x in range(w):
        queue.append((x, 0))
        queue.append((x, h-1))
    for y in range(h):
        queue.append((0, y))
        queue.append((w-1, y))
        
    def is_bg(r, g, b, a):
        # Allow slight off-white
        return r > 240 and g > 240 and b > 240 and a > 0
        
    bg_pixels = []
    
    while queue:
        x, y = queue.pop(0)
        if x < 0 or x >= w or y < 0 or y >= h:
            continue
        if (x, y) in visited:
            continue
        visited.add((x, y))
        
        r, g, b, a = pixels[x, y]
        if is_bg(r, g, b, a):
            bg_pixels.append((x, y))
            queue.append((x+1, y))
            queue.append((x-1, y))
            queue.append((x, y+1))
            queue.append((x, y-1))
            
    for x, y in bg_pixels:
        pixels[x, y] = (255, 255, 255, 0)
        
    # Also crop tight bounds
    return image.crop(image.getbbox())

for i, cat_img in enumerate(cats):
    trans_img = make_transparent(cat_img)
    out_path = os.path.join(out_dir, names[i])
    trans_img.save(out_path)
    print(f"Saved {out_path} - Size: {trans_img.size}")

File: split_cats.js
const Jimp = require('jimp');
const path = require('path');
const fs = require('fs');

const inputPath = '/Users/simbosung/Downloads/파일이름.png';
const outDir = path.join(__dirname, 'public', 'emotions');

if (!fs.existsSync(outDir)) {
  fs.mkdirSync(outDir, { recursive: true });
}

async function processImage() {
  try {
    console.log('Loading image...');
    const img = await Jimp.read(inputPath);
    
    const w = img.bitmap.width;
    const h = img.bitmap.height;
    const h1 = Math.floor(h / 3);
    
    const cats = [
      { name: 'angry.png', y: 0, height: h1 },
      { name: 'sad.png', y: h1, height: h1 },
      { name: 'happy.png', y: h1 * 2, height: h - (h1 * 2) }
    ];
    
    for (const cat of cats) {
      console.log(`Processing ${cat.name}...`);
      const catImg = img.clone().crop(0, cat.y, w, cat.height);
      
      // BFS flood fill from edges to make near-white background transparent
      const visited = new Set();
      const queue = [];
      
      // Add edges
      for (let x = 0; x < w; x++) {
        queue.push({x, y: 0});
        queue.push({x, y: cat.height - 1});
      }
      for (let y = 0; y < cat.height; y++) {
        queue.push({x: 0, y});
        queue.push({x: w - 1, y});
      }
      
      function isBg(color) {
        const rgba = Jimp.intToRGBA(color);
        return rgba.r > 240 && rgba.g > 240 && rgba.b > 240 && rgba.a > 0;
      }
      
      const bgPixels = [];
      let idx = 0;
      
      while (idx < queue.length) {
        const {x, y} = queue[idx++];
        if (x < 0 || x >= w || y < 0 || y >= cat.height) continue;
        
        const key = `${x},${y}`;
        if (visited.has(key)) continue;
        visited.add(key);
        
        const color = catImg.getPixelColor(x, y);
        if (isBg(color)) {
          bgPixels.push({x, y});
          queue.push({x: x+1, y});
          queue.push({x: x-1, y});
          queue.push({x, y: y+1});
          queue.push({x, y: y-1});
        }
      }
      
      console.log(`Found ${bgPixels.length} background pixels for ${cat.name}`);
      for (const p of bgPixels) {
        catImg.setPixelColor(0x00000000, p.x, p.y);
      }
      
      catImg.autocrop();
      
      const outPath = path.join(outDir, cat.name);
      await catImg.writeAsync(outPath);
      console.log(`Saved ${outPath}`);
    }
    
    console.log('All done!');
    
  } catch (err) {
    console.error('Error processing image:', err);
  }
}

processImage();
